functional/install_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package functional
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/coreos/fleet/functional/platform"
+	"github.com/coreos/fleet/functional/util"
+)
+
+// Load service and discovery units and test whether discovery unit adds itself as a dependency for the service.
+func TestInstallUnit(t *testing.T) {
+	cluster, err := platform.NewNspawnCluster("smoke")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cluster.Destroy()
+
+	// Start with a two-nodes cluster
+	members, err := platform.CreateNClusterMembers(cluster, 2)
+	if err != nil {
+		t.Fatal(err)
+	}
+	m0 := members[0]
+	machines, err := cluster.WaitForNMachines(m0, 2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Ensure we can SSH into machine using fleetctl
+	for _, machine := range machines {
+		if stdout, stderr, err := cluster.Fleetctl(m0, "--strict-host-key-checking=false", "ssh", machine, "uptime"); err != nil {
+			t.Fatalf("Unable to SSH into fleet machine: \nstdout: %s\nstderr: %s\nerr: %v", stdout, stderr, err)
+		}
+	}
+
+	// Load unit files
+	stdout, stderr, err := cluster.Fleetctl(m0, "load", "fixtures/units/hello.service", "fixtures/units/discovery.service")
+	if err != nil {
+		t.Fatalf("Failed loading unit files: \nstdout: %s\nstderr: %s\nerr: %v", stdout, stderr, err)
+	}
+
+	checkActiveState := func() bool {
+		stdout, _, err := cluster.Fleetctl(m0, "--strict-host-key-checking=false", "ssh", "discovery.service", "systemctl show --property=ActiveState discovery.service")
+		if err != nil {
+			t.Error("Failed getting info using remote systemctl")
+		}
+		stdout = strings.TrimSpace(stdout)
+		if stdout == "ActiveState=active" {
+			return true
+		} else {
+			return false
+		}
+	}
+	checkInactiveState := func() bool {
+		stdout, _, err := cluster.Fleetctl(m0, "--strict-host-key-checking=false", "ssh", "discovery.service", "systemctl show --property=ActiveState discovery.service")
+		if err != nil {
+			t.Error("Failed getting info using remote systemctl")
+		}
+		stdout = strings.TrimSpace(stdout)
+		if stdout == "ActiveState=inactive" {
+			return true
+		} else {
+			return false
+		}
+	}
+
+	// Verify that discovery.service unit is stopped
```

Comment:
I think this comment is a bit misleading here: it sounds as if we are waiting for an active unit to become inactive -- which is the case for the other invocation later on, but not here.